lucene/core/src/test/org/apache/lucene/index/TestIndexWriter.java

```diff
@@ -4258,4 +4258,47 @@ public void testPendingNumDocs() throws Exception {
       }
     }
   }
+
+  public void testIndexWriterBlocksOnStall() throws IOException, InterruptedException {
+    try (Directory dir = newDirectory()) {
+      try (IndexWriter writer = new IndexWriter(dir, newIndexWriterConfig())) {
```

Comment:
pull it up into the same try (semicolon-separated)?

Reply:
it's more clear that way what is closed first which is important here. That's why I do it this way.